### PR TITLE
[WorkspaceTests] Create mocking infra for creating manifest graphs

### PR DESCRIPTION
**Pull request by aciidb0mb3r**
This removes a lot of boilerplate needed for workspace tests.

### CODE REVIEW

File: Tests/CommandsTests/WorkspaceTests.swift [hunk 1]
```diff
@@ -127,75 +127,40 @@ final class WorkspaceTests: XCTestCase {
...
-                        .Package(url: repos["A"]!.url, majorVersion: 1),
-                        .Package(url: "//B", majorVersion: 1)
+            let graph = try MockManifestGraph(at: path,
+                rootDeps: [
+                    MockDependency("A", version: v1),
+                    MockDependency("B", version: v1),
+                ],
+                packages: [
+                    MockPackage("A", version: v1, dependencies: [
+                        MockDependency("AA", version: v1)
```
Comment (ddunbar) [inline Tests/CommandsTests/WorkspaceTests.swift]: How does this infrastructure work if you want to supply multiple manifest versions?

Comment (ddunbar) [inline Tests/CommandsTests/WorkspaceTests.swift]: Also... it is a little unfortunate we have to maintain this piece and then a separate piece for managing the Git repositories. I wonder if there is a way to merge those things (maybe a custom repo provider?)?

Comment (aciidb0mb3r) [inline Tests/CommandsTests/WorkspaceTests.swift]: You just specify them in packages like. This is like a repo provider right?
```swift
packages: [
    MockPackage("A", version: v1),
    MockPackage("A", version: "1.0.1"),
]
```

Comment (aciidb0mb3r) [inline Tests/CommandsTests/WorkspaceTests.swift]: ... we could however reuse provider objects

Comment (ddunbar) [inline Tests/CommandsTests/WorkspaceTests.swift]: Ok, that was my question.

Comment (ddunbar) [inline Tests/CommandsTests/WorkspaceTests.swift]: Ok Im fine starting with this

**Comment by aciidb0mb3r** [comment]
@swift-ci please test

**Pull request merged**